sbncode/LArRecoProducer/TrackStoppingChi2Alg.cc

```diff
@@ -0,0 +1,83 @@
+#include "sbncode/LArRecoProducer/TrackStoppingChi2Alg.h"
+
+#include "TF1.h"
+#include "TGraph.h"
+
+
+sbn::TrackStoppingChi2Alg::TrackStoppingChi2Alg(fhicl::ParameterSet const& p) :
+  fFitRange(p.get<float>("FitRange"))
+  , fMaxdEdx(p.get<float>("MaxdEdx"))
+  , fMinHits(p.get<unsigned int>("MinHits"))
+{
+}
+
+sbn::StoppingChi2Fit sbn::TrackStoppingChi2Alg::RunFit(const std::vector<float> &dEdxVec, const std::vector<float> &resRangeVec) const
+{
+  if (dEdxVec.size() != resRangeVec.size())
+    throw cet::exception("TrackStoppingChi2Alg") << "dEdx and Res Range do not have same length: " << dEdxVec.size() << " and " << resRangeVec.size() << std::endl;
+
+  if (dEdxVec.size() < fMinHits)
+    return StoppingChi2Fit();
+
+  const auto graph(std::make_unique<TGraph>(dEdxVec.size(), &resRangeVec[0], &dEdxVec[0]));
+
+  // Try and fit a flat polynomial
+  graph->Fit("pol0", "Q");
+  const TF1* polFit = graph->GetFunction("pol0");
+  const float pol0Chi2(polFit ? polFit->GetChisquare() : -5.f);
+  const float pol0Fit(polFit ? polFit->GetParameter(0) : -5.f);
+
+  // Try to fit an exponential
+  graph->Fit("expo", "Q");
+  const TF1* expFit = graph->GetFunction("expo");
+  const float expChi2(expFit ? expFit->GetChisquare() : -5.f);
```

Comment:
Perosnally I don't trust TGraph/TF1 etc and would just code this manually. `pol0` fit at least is simply a mean. And isn't expo just a linear fit of the log values? I guess if it works it works...

Reply:
So this is actually from the original tool, I have just moved it into the alg so its highlighting as added lines. It seems to work for now. I think this would be a separate PR to amend this :)